### ISSUE TITLE
subkey yields inconsistent results with hard/soft mixed

### ISSUE THREAD

**Issue by jacogr** [paritytech/substrate#2001]
`cargo run --release -- --sr25519 inspect "bottom drive obey lake curtain smoke basket hold race lonely fit walk/foo//bar"`

Run 1 -

```
  Public key (hex): 0xaaf0fac3917a39680a9e37ee3a4dd20dd6ee1c621154e0906ed809ec22291578
  Address (SS58): 5FvqanwTCeSj61C11jMt61qYxmNLaJVAyJvCa6TsmSCu9WGc
```

Run 2 -

```
  Public key (hex): 0x961266aa6653385c330c331c04d338d0062c72e4cc66f6dfb2a9fe2e3b4cd60c
  Address (SS58): 5FTUWJpxwVFX5NBdV1e7JqUUufH2i2jDrC3BUnkCJRBSw4TB
```

Further inspection reveals -

- soft/soft (e.g. /foo/bar) and hard/hard (e.g. //foo//bar) yields consistent results
- hard/soft (e.g. //foo/bar) is consistent as well
- soft/hard (e.g. /foo//bar) is inconsistent.

This could be related to the implementation of https://github.com/paritytech/substrate/blob/master/core/primitives/src/sr25519.rs#L368 and the use thereof here https://github.com/paritytech/substrate/blob/master/core/primitives/src/sr25519.rs#L454 where the full secret instead of the seed is passed in.

Basically the secretKey has a secret and nonce part, where the nonce part is non-consistent when derived from seed. (Passing in the seed portion here, when we get a seed back as well, would make it consistent)

**Comment by burdges** [comment]
This is why I added some `hard_derive_mini_secret_key` functions that do not hash the nonce seed.

We can make the nonce seed deterministic of course, but there is no reason why someone should not change the nonce seed whenever they like, so depending upon it being constant sounds unwise.
